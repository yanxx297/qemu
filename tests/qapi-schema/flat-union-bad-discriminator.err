tests/qapi-schema/flat-union-bad-discriminator.json:11: Flat union 'TestUnion' discriminator must be a string
