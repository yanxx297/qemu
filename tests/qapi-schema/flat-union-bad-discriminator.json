# FIXME: we should require the discriminator to be a string naming a base-type member
{ 'enum': 'TestEnum',
  'data': [ 'value1', 'value2' ] }
{ 'type': 'TestBase',
  'data': { 'enum1': 'TestEnum', 'kind': 'str' } }
{ 'type': 'TestTypeA',
  'data': { 'string': 'str' } }
{ 'type': 'TestTypeB',
  'data': { 'integer': 'int' } }
{ 'union': 'TestUnion',
  'base': 'TestBase',
  'discriminator': [],
  'data': { 'kind1': 'TestTypeA',
            'kind2': 'TestTypeB' } }
