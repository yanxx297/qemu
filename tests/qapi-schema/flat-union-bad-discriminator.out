[OrderedDict([('enum', 'TestEnum'), ('data', ['value1', 'value2'])]),
 OrderedDict([('type', 'TestBase'), ('data', OrderedDict([('enum1', 'TestEnum'), ('kind', 'str')]))]),
 OrderedDict([('type', 'TestTypeA'), ('data', OrderedDict([('string', 'str')]))]),
 OrderedDict([('type', 'TestTypeB'), ('data', OrderedDict([('integer', 'int')]))]),
 OrderedDict([('union', 'TestUnion'), ('base', 'TestBase'), ('discriminator', []), ('data', OrderedDict([('kind1', 'TestTypeA'), ('kind2', 'TestTypeB')]))])]
[{'enum_name': 'TestEnum', 'enum_values': ['value1', 'value2']},
 {'enum_name': 'TestUnionKind', 'enum_values': None}]
[OrderedDict([('type', 'TestBase'), ('data', OrderedDict([('enum1', 'TestEnum'), ('kind', 'str')]))]),
 OrderedDict([('type', 'TestTypeA'), ('data', OrderedDict([('string', 'str')]))]),
 OrderedDict([('type', 'TestTypeB'), ('data', OrderedDict([('integer', 'int')]))])]
